spec/rspec/support/differ_spec.rb

```diff
@@ -261,6 +261,35 @@ def inspect; "<BrokenObject>"; end
           expect(diff).to be_diffed_as(expected_diff)
         end
 
+        it "outputs unified diff message of two hashes with BigDecimal keys" do
+          in_sub_process_if_possible do
+            require 'bigdecimal'
+            expect {
+              diff = differ.diff({ BigDecimal('3') => 'b'}, { 3 => 'b' })
+              if expect(diff).to match(/3.0 \(#<BigDecimal/)
+                raise StandardError, 'Passed Test'
+              end
+            }.to raise_error(StandardError, 'Passed Test')
```

Comment:
I guess when I wrote this, I didn't quite fully understand how `in_sub_process_if_possible` worked, as was under the impression that it only applied to the contents of what was inside `expects` with a block.

I agree that switching these tests to use time objects might be a better option. I decided to contribute to this repo initially as I wanted to change how BigDecimals outputted, but because we are outputting the inspect block from them, it gets all finicky when trying to match on the objects id.